va/va_test.go

```diff
@@ -278,6 +279,28 @@ func TestPerformValidationValid(t *testing.T) {
 	if !strings.Contains(resultLog[0], `"Hostname":"good-dns01.com"`) {
 		t.Errorf("PerformValidation didn't log validation hostname.")
 	}
+
+	// Regex to detect if there is a validated field and capture
+	// everything in the value.
+	valRegex, err := regexp.Compile(`"validated":"(.+?)"`)
+	if err != nil {
+		t.Errorf("Could not create regexp.Compile for valTimestamp.")
+	}
```

Comment:
For tests, `regex.MustCompile` is suitable and saves a few lines.

However, now that we're getting into more complicated checks than the simple `strings.Contains` we had before, it might be more suitable to extract the JSON part of this log line, parse it as JSON, then check specific fields.

Reply:
I played a bit with what it would look like to regex match and extract the JSON in the log line then unmarshal that to something and check the fields. This seemed cleaner and a more reliable match in case that the test log line was ever modified by adding non JSON to the end of it. Also, there is only one other working string.Contains match in this test that would benefit from the change.

I am happy to put up something if there is a strong preference for parsing the JSON though.

Reply:
That makes sense that extracting the JSON is complex.

Looking at this again, and putting it in context with Aaron's comments about using the fakeclock: You can actually turn this test into a much simpler `.Contains` test, while also making it a stricter test. Now that you're using fakeclock, we should always get the exact same validatation time, and you can check against that exact string rather than just checking that we could parse it as a date.

Reply:
Added a fake clock and a check using .Contains instead of regexp.